src/diffusers/models/prior_transformer.py

```diff
@@ -59,24 +59,50 @@ def __init__(
         num_embeddings=77,
         additional_embeddings=4,
         dropout: float = 0.0,
+        time_embed_act_fn: str = "silu",
+        norm_embedding_proj: bool = False,
+        time_embed_dim: Optional[int] = None,
+        clip_embedding_dim: Optional[int] = None,
+        out_dim: Optional[int] = None,
     ):
         super().__init__()
         self.num_attention_heads = num_attention_heads
         self.attention_head_dim = attention_head_dim
         inner_dim = num_attention_heads * attention_head_dim
         self.additional_embeddings = additional_embeddings
 
+        if time_embed_dim is None:
+            time_embed_dim = inner_dim
+
+        if clip_embedding_dim is None:
+            clip_embedding_dim = embedding_dim
+
+        if out_dim is None:
+            out_dim = embedding_dim
+
         self.time_proj = Timesteps(inner_dim, True, 0)
-        self.time_embedding = TimestepEmbedding(inner_dim, inner_dim)
+        self.time_embedding = TimestepEmbedding(inner_dim, time_embed_dim, out_dim=inner_dim, act_fn=time_embed_act_fn)
 
         self.proj_in = nn.Linear(embedding_dim, inner_dim)
 
-        self.embedding_proj = nn.Linear(embedding_dim, inner_dim)
-        self.encoder_hidden_states_proj = nn.Linear(embedding_dim, inner_dim)
+        if norm_embedding_proj:
+            self.embedding_proj_norm = nn.LayerNorm(clip_embedding_dim)
+        else:
+            self.embedding_proj_norm = None
+
+        self.embedding_proj = nn.Linear(clip_embedding_dim, inner_dim)
+
+        if self.config.clip_embedding_dim is None:
+            self.encoder_hidden_states_proj = nn.Linear(embedding_dim, inner_dim)
+        else:
+            self.encoder_hidden_states_proj = None
```

Comment:
I see how this can save new config params, but here the logic is not sound enough I think. Let's add a new `encoder_hid_proj: Optional[str] = "linear"` to the init that we set to `None` for Shape-E.

```suggestion
        if encoder_hid_proj is not None:
            self.encoder_hidden_states_proj = nn.Linear(embedding_dim, inner_dim)
        else:
            self.encoder_hidden_states_proj = None
```